include/cm/nccl_ofi_cm_comms.h

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2025 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
+
+#ifndef NCCL_OFI_CM_COMMS_H_
+#define NCCL_OFI_CM_COMMS_H_
+
+#include <deque>
+#include <optional>
+
+#include "cm/nccl_ofi_cm_types.h"
+#include "cm/nccl_ofi_cm_reqs.h"
+#include "nccl_ofi_freelist.h"
+
+/**
+ * A handle returned by connection_manager::listen(), used to accept
+ * incoming connections from the listen() and create recv handles
+ */
+class nccl_ofi_cm_l_comm
+{
+public:
+	/**
+	 * Obtain the handle associated with this cm_l_comm
+	 *
+	 * The caller (transport) should return this handle back to NCCL to
+	 * be delivered out-of-band to the remote (send side) node
+	 */
+	nccl_net_ofi_conn_handle get_handle() { return handle; }
+
+	/**
+	 * Accept an incoming connect message from the send side to this handle
+	 * This returns a cm_r_comm that can be used to create the protocol-specific
+	 * communicator
+	 */
+	nccl_ofi_cm_r_comm *accept();
+
+	/* --------------------------------------------------------- */
+	/* Not for use by transport code */
+
+	/**
+	 * Construct a new cm_l_comm
+	 *
+	 * Not to be called by the transport code -- the transport code
```

Comment:
Some `protected` and `friend` keywords seem way better than a comment here.  And again with the pointers.